training/deepspeech_training/util/sample_collections.py

```diff
@@ -483,6 +499,9 @@ def samples_from_sources(sample_sources, buffering=BUFFER_SIZE, labeled=None):
     if len(sample_sources) == 0:
         raise ValueError('No files')
     if len(sample_sources) == 1:
-        return samples_from_source(sample_sources[0], buffering=buffering, labeled=labeled)
-    cols = list(map(partial(samples_from_source, buffering=buffering, labeled=labeled), sample_sources))
-    return Interleaved(*cols, key=lambda s: s.duration)
+        return samples_from_source(sample_sources[0], buffering=buffering, labeled=labeled, reverse=reverse)
+    cols = list(map(partial(samples_from_source,
+                            buffering=buffering,
+                            labeled=labeled,
+                            reverse=reverse), sample_sources))
+    return Interleaved(*cols, key=lambda s: -s.duration if reverse else s.duration)
```

Comment:
FWIW, `heapq.merge` also has a `reverse` parameter if you want to keep the same pattern as the rest of the PR, but either way is fine for me.